src/apply.jl

```diff
@@ -314,13 +314,13 @@ function ITensors.apply(
 
     for vn in neighbors(ψ, src(e))
       if (vn != dst(e))
-        ψv1 = noprime(ψv1 * bond_tensors[vn => src(e)])
+        ψv1 = noprime(ψv1 * bond_tensors[NamedEdge(vn => src(e))])
```

Comment:
It's kind of unfortunate that we need to add the `NamedEdge` wrapper now.

Maybe we could use a `DataGraph` for the cache but just keep it simple and only have edge data?